### PR TITLE
Kol opt tweet search

### PR DESCRIPTION
**Pull request by Xm0onh**
Adding more optimization toward searching tweets of KOLs
* Select randomly `ACCOUNTS_PER_BATCH` accounts and search for `MAX_SEARCH_TWEETS` of their recent tweets. Also there is a further shrinking to make sure less-engaged accounts would get a chance to be seen.
* optimize timeline fetcher -> scraping controllable number of tweets, `MAX_TIMELINE_TWEETS`
* add check to avoid of adding duplicated tweet to vec db

### CODE REVIEW

File: auto-kol/agent/src/tools/tools/tweetSearchTool.ts [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 const logger = createLogger('tweet-search-tool');
 
+
+
+function getRandomAccounts(accounts: string[], n: number): string[] {
+    const shuffled = [...accounts].sort(() => 0.5 - Math.random());
+    return shuffled.slice(0, Math.min(n, accounts.length));
+}
+
 export const createTweetSearchTool = (scraper: any) => new DynamicStructuredTool({
```
Comment (jfrank-summit) [inline auto-kol/agent/src/tools/tools/tweetSearchTool.ts:16]: Let's add `Scraper` type here.

File: auto-kol/agent/src/tools/tools/tweetSearchTool.ts [hunk 2]
```diff
@@ -17,8 +23,8 @@ export const createTweetSearchTool = (scraper: any) => new DynamicStructuredTool
         try {
             logger.info('Called search_recent_tweets');
             await updateKOLs();
-            const cleanAccounts = await getKOLsAccounts()
-            logger.info('Fetching KOLs:', await getKOLsAccounts());
+            const cleanAccounts = await getKOLsAccounts();
```
Comment (jfrank-summit) [inline auto-kol/agent/src/tools/tools/tweetSearchTool.ts:26]: I'm not sure what `cleanAccounts` means, is there a more descriptive name?

File: auto-kol/agent/src/utils/twitter.ts [hunk 3]
```diff
@@ -8,6 +8,8 @@ import { Tweet } from '../types/twitter.js';
 
 const logger = createLogger('twitter-utils');
 const twitterScraper = await createTwitterClientScraper();
+
+
```
Comment (jfrank-summit) [inline auto-kol/agent/src/utils/twitter.ts:12]: Too many lines
```suggestion
```

File: auto-kol/agent/src/tools/tools/tweetSearchTool.ts (hunk 1 above)
Comment (Xm0onh) [reply auto-kol/agent/src/tools/tools/tweetSearchTool.ts]: I added in the root: 
`export const createTools = (scraper: ExtendedScraper) => {`
Also each tool that is using scraper.

File: auto-kol/agent/src/config/index.ts [hunk 4]
```diff
@@ -46,4 +46,9 @@ export const config = {
     RPC_URL: process.env.RPC_URL,
     CONTRACT_ADDRESS: process.env.CONTRACT_ADDRESS,
     PRIVATE_KEY: process.env.PRIVATE_KEY,
-}; 
\ No newline at end of file
+
+    // Tweet Search/Fetch Configuration
+    ACCOUNTS_PER_BATCH: 10,
+    MAX_SEARCH_TWEETS: 20,
+    MAX_TIMELINE_TWEETS: 10,
```
[the remark this answers is not in the document]
Comment (Xm0onh) [reply auto-kol/agent/src/config/index.ts:53]: Just a parameter :) 
I didn't want to push it much as I need to stay under radar of getting throttled for further testing lol

[the remark this answers is not in the document]
Comment (jfrank-summit) [reply auto-kol/agent/src/config/index.ts:53]: Maybe we should move to env variables so that they are more persistent across code updates.

**Review by jfrank-summit — APPROVED** [review]

**Pull request merged**